docs/devguide/flashfirmware.rst

```diff
@@ -1,58 +1,83 @@
 .. _flashfirmware:
 
-=================
-Flashing Firmware
-=================
+========================
+Build and flash firmware
+========================
 
-Building firmware
------------------
-Use yotta to build.
+micro:bit V1
+============
 
-Use target bbc-microbit-classic-gcc-nosd::
+This applies to MicroPython for the micro:bit V1, the source of which can be 
+found here: `bbcmicrobit/micropython <https://github.com/bbcmicrobit/micropython>`_.
 
-  yt target bbc-microbit-classic-gcc-nosd
+Dependencies
+------------
+- `yotta <http://docs.yottabuild.org//>`_
 
-Run yotta update to fetch remote assets::
+When installing `yotta <http://docs.yottabuild.org//>`_, make sure you have
+these components ticked to install.
 
-  yt up
+- python
+- gcc
+- cMake
+- ninja
+- Yotta
+- git-scm
+- mbed serial driver
+
+The `yotta
+<http://docs.yottabuild.org//>`_ tool is used to build MicroPython, but before that takes place
+additional files have to be generated by the Makefile in preparation for the 
+build, and additional data is added to the hex file after.
+
+Use yotta to build::
 
-Start the build with either yotta::
+  yt target bbc-microbit-classic-gcc@https://github.com/lancaster-university/yotta-target-bbc-microbit-classic-gcc
 
-  yt build
+Run yotta update to fetch remote assets::
+
+  yt up
 
-...or use the Makefile::
+The yotta tool is used to build MicroPython, but before that takes place
+additional files have to be generated by the Makefile in preparation for the 
+build, and additional data is added to the hex file after.
 
-  make all
+Running the ``make all`` command executes the following steps:
 
-The result is a microbit-micropython hex file (i.e. ``microbit-micropython.hex``)
-found in the build/bbc-microbit-classic-gcc-nosd/source from the root of the
-repository.
+- The ``tools/makeversionhdr.py`` script creates the ``microbitversion.h`` file 
+  with macros containing build information
+- Yotta builds the source and creates a bare hex file with just the firmware
+- The ``tools/adduicr.py`` script adds the UICR to the bare hex
+- The final hex file is placed in ``build/firmware.hex``
+- The user can optionally append a script using ``tools/makecombinedhex.py`` 
+  (or other tools)
 
-The Makefile does some extra preprocessing of the source, which is needed only
-if you add new interned strings to ``qstrdefsport.h``. The Makefile also puts
-the resulting firmware at build/firmware.hex, and includes some convenience
-targets.
+micro:bit V2
+============
 
-Preparing firmware and a Python program
```

Comment:
This section was still useful for MicroPython v1, we can just indicate the commands needed to ru makecombined

Reply:
Done.